src/main/kotlin/no/nav/syfo/client/HttpClientCommon.kt

```diff
@@ -2,47 +2,35 @@ package no.nav.syfo.client
 
 import io.ktor.client.plugins.contentnegotiation.*
 import io.ktor.client.*
+import io.ktor.client.engine.*
 import io.ktor.client.engine.apache.*
 import io.ktor.client.plugins.*
 import io.ktor.serialization.jackson.*
 import no.nav.syfo.util.configure
 import org.apache.http.impl.conn.SystemDefaultRoutePlanner
 import java.net.ProxySelector
 
-val config: HttpClientConfig<ApacheEngineConfig>.() -> Unit = {
+val commonConfig: HttpClientConfig<out HttpClientEngineConfig>.() -> Unit = {
     install(ContentNegotiation) {
         jackson { configure() }
     }
-    expectSuccess = false
-}
-
-val httpClientWithProxyAndTimeout: HttpClientConfig<ApacheEngineConfig>.() -> Unit = {
-    install(ContentNegotiation) {
-        jackson { configure() }
-    }
-    expectSuccess = true
-    install(HttpTimeout) {
-        socketTimeoutMillis = 60000
-    }
-    engine {
-        customizeClient {
-            setRoutePlanner(SystemDefaultRoutePlanner(ProxySelector.getDefault()))
+    install(HttpRequestRetry) {
+        retryOnExceptionIf(2) { _, cause ->
+            cause !is ClientRequestException
         }
+        constantDelay(500L)
     }
+    expectSuccess = true
```

Comment:
Prøver å forstå her, men er greia at man setter en overordnet retry-strategi, i stedet for i hver eneste client?

`retryOnExceptionIf(2)`, hva betyr det? Retry 2 ganger?

Reply:
Ja, dette er retry-mekanismen vi har tatt i bruk i de andre app'ene. Her i padm2 erstatter det et hjemmesnekret opplegg (NetworksHelpers.kt). 

Slik det er konfigurert forsøker den på nytt 2 ganger (avhengig av hva slags exception som kastes).